sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -132,12 +131,11 @@ case class DataSource(
               }
           }
         } catch {
-          case e: NoClassDefFoundError => // This one won't be caught by Scala NonFatal
-            // NoClassDefFoundError's class name uses "/" rather than "." for packages
-            val className = e.getMessage.replaceAll("/", ".")
-            if (spark2RemovedClasses.contains(className)) {
-              throw new ClassNotFoundException(s"$className was removed in Spark 2.0. " +
-                "Please check if your library is compatible with Spark 2.0", e)
+          case e: NoClassDefFoundError =>
+            // e.getMessage is the class name of provider2. Instead, we use provider here.
+            if (spark2RemovedClasses.contains(provider)) {
```

Comment:
In my previous PR, I want to provide a better message for e.g., `org.apache.spark.sql.DataFrame` not found. It usually happens when calling some method (has a `org.apache.spark.sql.DataFrame` parameter) in a class that is compiled with an old Spark. Obviously, here `provider` won't be `org.apache.spark.sql.DataFrame`.